### PR TITLE
Add Source Code Integration param and SAM usage instructions

### PR DESCRIPTION
**Pull request by sfirrin**
### What does this PR do?
Adds a param to the macro that can be used with SAM's --parameter-overrides option and local git commands to automatically add the `git.commit.sha` and `git.repository_url` tags used by Datadog Source Code Integration. 

### Motivation
Easy set up and usage of SCI

### Testing Guidelines
Deployed with a sample SAM app with and without setting the param. 

### Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [x] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [x] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: serverless/src/git.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    if (repositoryUrl.startsWith("git@")) {
+      return repositoryUrl;
+    }
+    const { protocol, hostname, pathname } = new URL(repositoryUrl);
+    if (!protocol || !hostname) {
+      return repositoryUrl;
+    }
+
+    return `${protocol}//${hostname}${pathname}`;
```
Comment (duncanista) [inline serverless/src/git.ts:14]: Do we have a test for this line?

File: serverless/src/git.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+export const getGitTagsFromParam = (gitDataParam: string) => {
+  const [commitSha, rawGitConfigUrlOutput] = gitDataParam.split(",", 2);
+
+  const sanitizedRemote = filterAndFormatGithubRemote(rawGitConfigUrlOutput);
+
+  const gitCommitShaTag = `git.commit.sha:${commitSha}`;
+  const gitRepoUrlTag = `git.repository_url:${sanitizedRemote}`;
+
+  return { gitCommitShaTag, gitRepoUrlTag };
+};
```
Comment (duncanista) [inline serverless/src/git.ts:41]:
```suggestion
/**
 * Removes sensitive information from the given git remote
 * URL and normalizes the URL prefix.
 * 
 * @example
 * // returns "github.com/"
 * filterAndFormatGithubRemote("git@github.com")
 * filterAndFormatGithubRemote("https://github.com")
 * 
 * @param rawRemote git remote URL.
 * @returns 
 */
```

Comment (duncanista) [reply serverless/src/git.ts:41]: Something happened here, but i just meant to update the JS Docs.

File: serverless/README.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ aws cloudformation update-stack \
 
 ## Usage with AWS SAM
 
-To deploy your serverless application with SAM, add the Datadog Serverless CloudFormation macro under the `Transform` section in your `template.yml` file, after the required SAM transform:
+To deploy your serverless application with SAM, add the Datadog Serverless CloudFormation macro under the `Transform` section in your `template.yml` file, after the required SAM transform. Also add a DDGitData param and pass it to the macro to enable Datadog Source Code Integration:
```
Comment (kayayarai) [inline serverless/README.md:45]:
```suggestion
To deploy your serverless application with SAM, add the Datadog Serverless CloudFormation macro under the `Transform` section in your `template.yml` file, after the required SAM transform. Also add a `DDGitData` parameter and pass it to the macro to enable Datadog Source Code Integration:
```

File: serverless/README.md [hunk 4]
```diff
@@ -57,7 +57,20 @@ Transform:
...
       # For additional parameters, see the Configuration section
+
+Parameters:
+  DDGitData:
+    Type: String
+    Default: ""
+    Description: "The output of $(git rev-parse HEAD),$(git config --get remote.origin.url). Used for Datadog Source Code Integration tagging"
+```
+
+To set the DDGitData parameter for Datadog's Source Code Integration, use SAM's `--parameter-overrides` option:
```
Comment (kayayarai) [inline serverless/README.md:71]:
```suggestion
To set the `DDGitData` parameter for Datadog's Source Code Integration, use SAM's `--parameter-overrides` option:
```

**Review by kayayarai — APPROVED** [review]
minor edit to docs and 👍

File: serverless/src/git.ts (hunk 1 above)
Comment (sfirrin) [reply serverless/src/git.ts:14]: Good question, I will add some tests for filterSensitiveInfoFromRepository

**Review by duncanista — APPROVED** [review]
LGTM :shipit:

**Pull request merged**